core/app/models/spree/stock_location.rb

```diff
@@ -12,8 +12,8 @@ class StockLocation < Spree::Base
     scope :active, -> { where(active: true) }
     scope :order_default, -> { order(default: :desc, name: :asc) }
 
-    after_create :create_stock_items, :if => "self.propagate_all_variants?"
     after_save :ensure_one_default
+    after_commit :create_stock_items, on: :create, if: "self.try :propagate_all_variants?"
```

Comment:
While it would be appropriate to move seed data to seeds.rb from migrations I'm not really sure the best way to do that to ensure people upgrading still get proper seed data.  We could also add the missing column to the migration running into the issue, and then skip adding the column in the second migration if it already exists.